pkg/controller/api/radapp.io/v1alpha3/deploymenttemplate_types.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2024 The Radius Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// DeploymentTemplateSpec defines the desired state of DeploymentTemplate
+type DeploymentTemplateSpec struct {
+	// Template is the ARM JSON manifest that defines the resources to deploy.
+	Template string `json:"template,omitempty"`
+
+	// Parameters is the ARM JSON parameters for the template.
+	Parameters map[string]string `json:"parameters,omitempty"`
+
+	// ProviderConfig specifies the scope for resources
+	ProviderConfig string `json:"providerConfig,omitempty"`
+
+	// RootFileName is the name of the Bicep file that
+	// `bicep build` is run on.
+	RootFileName string `json:"rootFileName,omitempty"`
+}
+
+// DeploymentTemplateStatus defines the observed state of DeploymentTemplate
+type DeploymentTemplateStatus struct {
+	// ObservedGeneration is the most recent generation observed for this DeploymentTemplate.
+	ObservedGeneration int64 `json:"observedGeneration,omitempty" protobuf:"varint,1,opt,name=observedGeneration"`
+
+	// Template is the ARM JSON manifest that defines the resources to deploy.
+	Template string `json:"template,omitempty"`
+
+	// Parameters is the ARM JSON parameters for the template.
+	Parameters string `json:"parameters,omitempty"`
+
+	// ProviderConfig specifies the scope for resources
+	ProviderConfig string `json:"providerConfig,omitempty"`
+
+	// RootFileName is the name of the Bicep file that
+	// `bicep build` is run on.
+	RootFileName string `json:"rootFileName,omitempty"`
```

Comment:
Same question as DeploymentResource, why do we duplicate the fields from spec into the status?